docs/querying/query-context.md

```diff
@@ -106,3 +106,4 @@ vectorization. These query types will ignore the "vectorize" parameter even if i
 |--------|-------|------------|
 |vectorize|`true`|Enables or disables vectorized query execution. Possible values are `false` (disabled), `true` (enabled if possible, disabled otherwise, on a per-segment basis), and `force` (enabled, and groupBy or timeseries queries that cannot be vectorized will fail). The `"force"` setting is meant to aid in testing, and is not generally useful in production (since real-time segments can never be processed with vectorized execution, any queries on real-time data will fail). This will override `druid.query.default.context.vectorize` if it's set.|
 |vectorSize|`512`|Sets the row batching size for a particular query. This will override `druid.query.default.context.vectorSize` if it's set.|
+|vectorizeVirtualColumns|`true`|Enables or disables vectorized query processing of queries with virtual columns, layered on top of `vectorize` (`vectorize` must also be set to true for a query to utilize vectorization). Possible values are `false` (disabled), `true` (enabled if possible, disabled otherwise, on a per-segment basis), and `force` (enabled, and groupBy or timeseries queries with virtual columns that cannot be vectorized will fail). The `"force"` setting is meant to aid in testing, and is not generally useful in production. This will override `druid.query.default.context.vectorizeVirtualColumns` if it's set.|
```

Comment:
IMO, it'd be best to set it to `false` now, then change it to `true` for the next release. It'll lower the risk around upgrading, and we want upgrades to be really low risk so people feel comfortable just dropping in the newest version.

The specific risks here are:

1) Bugs in vectorized expressions.
2) Situations where vectorized expressions perform worse than non-vectorized ones (perhaps some optimizations are implemented for the non-vectorized case, but not the vectorized case).
3) Situations where there is a bug in the vectorized query engines _in general_, but a specific user isn't hitting it, because they're using expressions and so the query wouldn't vectorize anyway.

Reply:
Given that #10429 has uncovered a bug with vectorized group by engine when grouping on numeric columns with null values in sql compatible null handling mode that isn't related to expressions/virtual columns, I guess this seems prudent and I've come around to being in the 'false by default' camp.

Reply:
updated to default to false. I modified the tests to also set this parameter to `force` whenever `vectorize` is set to `force` so that we can keep the test coverage